src/server.ts

```diff
@@ -135,13 +143,21 @@ export async function startPluginsServer(
         ;[server, closeServer] = await createServer(serverConfig, null)
 
         piscina = makePiscina(serverConfig)
-        const processEvent = (event: PluginEvent) => piscina!.runTask({ task: 'processEvent', args: { event } })
+        const processEvent = (event: PluginEvent) => {
+            if ((piscina?.queueSize || 0) > (server?.WORKER_CONCURRENCY || 4) * (server?.WORKER_CONCURRENCY || 4)) {
+                queue?.pause()
+            }
+            return piscina!.runTask({ task: 'processEvent', args: { event } })
+        }
 
         if (!server.DISABLE_WEB) {
             fastifyInstance = await startFastifyInstance(server)
         }
 
         queue = startQueue(server, processEvent)
+        piscina.on('drain', () => {
+            queue?.resume()
+        })
```

Comment:
This should be in startQueue, since we'll want different `drain` handlers in the Redis and Kafka `startQueue`s.

Reply:
Hmm... sure about that? The `pause` code is just a few lines above and I think just getting `pause`/`resume` called on it is all the queue needs to know... it shouldn't really know much about what is `piscina` I think.

I'll merge this for now and we can refactor this when merging the Kafka PR.